orientdb/src/test/java/org/hibernate/ogm/datastore/orientdb/utils/OrientDBTestHelper.java

```diff
@@ -152,45 +134,76 @@ public boolean backendSupportsTransactions() {
+				.concat( propertyReader.property( OrientDBProperties.DATETIME_FORMAT, String.class ).withDefault( "yyyy-MM-dd HH:mm:ss" ).getValue() )
+				.concat( "'" ) );
+
 	}
 
 	@Override
 	public void dropSchemaAndDatabase(SessionFactory sessionFactory) {
 		OrientDBDatastoreProvider provider = getProvider( sessionFactory );
-		OrientJdbcConnection connection = (OrientJdbcConnection) provider.getConnection();
-		log.infof( "call dropSchemaAndDatabase! db closed: %b ", connection.getDatabase().isClosed() );
-		if ( !connection.getDatabase().isClosed() ) {
-			ODatabaseDocumentTx db = (ODatabaseDocumentTx) connection.getDatabase();
-			if ( !db.isActiveOnCurrentThread() ) {
-				db.activateOnCurrentThread();
-			}
-			if ( db.getTransaction().isActive() ) {
-				db.getTransaction().close();
-			}
-			MemoryDBUtil.dropInMemoryDb();
+		ConfigurationPropertyReader propertyReader = provider.getPropertyReader();
+		OrientDBProperties.DatabaseTypeEnum databaseType = propertyReader
+				.property( OrientDBProperties.DATEBASE_TYPE, OrientDBProperties.DatabaseTypeEnum.class )
+				.withDefault( OrientDBProperties.DatabaseTypeEnum.MEMORY ).getValue();
+		ODatabaseDocumentTx db = provider.getCurrentDatabase();
+		log.infof( "call dropSchemaAndDatabase! db closed: %b ", db.isClosed() );
+
+		Set<String> docClassSet = new HashSet<>();
+		docClassSet.add( "sequences" );
+
+		// remove all documents and classes
+		log.info( "try to delete all documents" );
+		List<ODocument> result = NativeQueryUtil.executeIdempotentQuery( db, "select from V" );
+		for ( ODocument doc : result ) {
+			log.infof( "doc: %s ", doc.toJSON() );
+			log.infof( "rid class: %s ", doc.field( "@rid" ).getClass().getName() );
+			log.infof( "class class: %s ", doc.field( "@class" ).getClass().getName() );
+			String docClass = doc.field( "@class", String.class );
+			ORecordId rid = doc.field( "@rid", ORecordId.class );
+			docClassSet.add( docClass );
+			db.delete( rid );
+		}
+		log.infof( "try to delete classes %s", docClassSet );
+		for ( String docClass : docClassSet ) {
+			NativeQueryUtil.executeIdempotentQuery( db, String.format( "select executeQuery('drop class %s')", docClass ) );
+		}
+		// remove all functions
+		log.info( "try to delete all functions" );
+		for ( ODocument func : NativeQueryUtil.executeIdempotentQuery( db, "select from OFunction" ) ) {
+			ORecordId rid = func.field( "@rid", ORecordId.class );
+			log.infof( "delete function by rid %s", rid.toString() );
+			db.delete( rid );
+		}
+		String user = propertyReader.property( OgmProperties.USERNAME, String.class ).getValue();
+		String password = propertyReader.property( OgmProperties.PASSWORD, String.class ).getValue();
+		String host = propertyReader.property( OgmProperties.HOST, String.class ).withDefault( "localhost" ).getValue();
+		String database = propertyReader.property( OgmProperties.DATABASE, String.class ).getValue();
+		if ( OrientDBProperties.DatabaseTypeEnum.REMOTE.equals( databaseType ) ) {
```

Comment:
Ok, for Remote we need the ServerAdmin, but in all the other cases can't we just use `db.drop()`?